tests/functional/test_cli.py

```diff
@@ -94,4 +106,52 @@ def test_run_hello_workflow(self, flavor):
             contents = fp.read()
 
         self.assertEquals(contents, self.expected_contents)
+        shutil.rmtree(self.scratch_dir)
+
+    @parameterized.expand([
+        ("request_through_stdin"),
+        ("request_through_argument")
+    ])
+    def test_run_hello_workflow_incompatible(self, flavor):
```

Comment:
This test is good for now, but it will need an upgrade when we change the version check logic or bump beyond 2.0.